src/main/java/me/senseiwells/essentialclient/clientscript/values/PlayerValue.java

```diff
@@ -189,10 +189,11 @@ public ArucasFunctionMap<MemberFunction> getDefinedMethods() {
 		)
 		private Value use(Arguments arguments) throws CodeError {
 			StringValue stringValue = arguments.skip().getNextString();
+			final MinecraftClient client = ArucasMinecraftExtension.getClient();
 			switch (stringValue.value.toLowerCase()) {
-				case "hold" -> ArucasMinecraftExtension.getClient().options.useKey.setPressed(true);
-				case "stop" -> ArucasMinecraftExtension.getClient().options.useKey.setPressed(false);
-				case "once" -> ((MinecraftClientInvoker) ArucasMinecraftExtension.getClient()).rightClickMouseAccessor();
+				case "hold" -> client.execute(()->client.options.useKey.setPressed(true));
+				case "stop" -> client.execute(()->client.options.useKey.setPressed(false));
+				case "once" -> client.execute(((MinecraftClientInvoker) client)::rightClickMouseAccessor);
```

Comment:
rightClickMouseAccessor and leftClickMouseAccessor area also already on the main thread